mariiaz2306/garden_store
# Refactored data fetching with RTK Query and optimized product display

Summary

This PR refactors data fetching and state management using RTK Query, simplifying API interactions. It also introduces logic for displaying four random discounted products on the homepage, enhancing user engagement.

Changes

Adopted RTK Query for API requests.
Removed obsolete async thunk and reducer code for products and categories.
Implemented RTK Query endpoints for fetching categories and products, including discounted items.
Updated components to utilize RTK Query hooks for data fetching.

Goals

Simplify and improve state management and API requests.
Improve code maintainability and readability.
Enhance UX with dynamic display of discounted products.

Testing

Ensured data fetching works correctly with RTK Query.
Locally tested the display of discounted products.
Verified no errors in the console and the app functions correctly after changes.

How to Test

Visit the homepage.
Confirm that categories and products are displayed correctly.
Check that four random discounted products are displayed on the homepage.
Navigate through the site to ensure stability.
